--- context.py	2019-01-18 17:36:03.000000000 +0100
+++ ~/.local/lib/python3.6/site-packages/zmq/sugar/context.py	2019-01-18 18:00:23.726179780 +0100
@@ -42,6 +42,7 @@
     
     def __del__(self):
         """deleting a Context should terminate it, without trying non-threadsafe destroy"""
+        print("Context.__del__")
         if not self._shadow and not _exiting:
             self.term()
     
@@ -49,6 +50,7 @@
         return self
     
     def __exit__(self, *args, **kwargs):
+        print("Context.__exit__")
         self.term()
     
     def __copy__(self, memo=None):
